src/controller/node.c

```diff
@@ -904,6 +961,18 @@ static int node_method_register(sd_bus_message *m, void *userdata, UNUSED sd_bus
                 return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_SERVICE_UNKNOWN, "Unexpected node name");
         }
 
+        if (named_node->required_selinux_context &&
+            (node->peer_selinux_context == NULL ||
+             !streq(node->peer_selinux_context, named_node->required_selinux_context))) {
+                bc_log_errorf("Node tried to register as '%s' with wrong selinux context '%s', expected '%s'",
+                              name,
+                              node->peer_selinux_context ? node->peer_selinux_context : "(missing)",
+                              named_node->required_selinux_context);
+
+                /* This is same as for not allowed, don't allow probing for available names */
+                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_SERVICE_UNKNOWN, "Unexpected node name");
```

Comment:
Could we use a different error type here? Maybe `SD_BUS_ERROR_AUTH_FAILED`?
```suggestion
                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_AUTH_FAILED, "Node using wrong SELinux context");
```

Reply:
I was intentionally trying to not leak information that there was a name like this that you were not allowed to connect as. This way an "attacker" could not probe for information like this. Not sure it is useful...

Reply:
Good point! I thought about having a typo in the config file like `japroxy_t` instead of `haproxy_t` and then trying to debug why the setup is not working. Maybe its fine to state that the context is wrong since the specific context name is not given.

Reply:
Well, the correct error is printed in the logs on the controller side. But sure, it can be confusing on the agent side.

Reply:
We sort of already leak already connected node names just below this (with the "already connected" thing) though, so maybe this isn't really a huge security issue.

Reply:
I'll make it SD_BUS_ERROR_ACCESS_DENIED